File: legacy/_1.10.2/src/main/java/cn/enaium/noexpensive/command/NoExpensiveCommand.java
package cn.enaium.noexpensive.command;

import cn.enaium.noexpensive.Config;
import net.minecraft.command.AbstractCommand;
import net.minecraft.command.CommandException;
import net.minecraft.command.CommandSource;
import net.minecraft.server.MinecraftServer;
import net.minecraft.text.LiteralText;
import net.minecraft.text.TranslatableText;

/**
 * @author Enaium
 */
public class NoExpensiveCommand extends AbstractCommand {

    @Override
    public String getCommandName() {
        return "noexpensive";
    }

    @Override
    public String getUsageTranslationKey(CommandSource source) {
        return null;
    }

    @Override
    public void method_3279(MinecraftServer minecraftServer, CommandSource commandSource, String[] args) throws CommandException {
        if (args.length == 0) {
            commandSource.sendMessage(new LiteralText("§cNoExpensive §7by §bEnaium"));
            return;
        }

        if (args[0].equalsIgnoreCase("reload")) {
            Config.load();
            commandSource.sendMessage(new TranslatableText("command.reload.success"));
        }
    }
}

File: legacy/_1.11.2/src/main/kotlin/cn/enaium/noexpensive/command/NoExpensiveCommand.kt
package cn.enaium.noexpensive.command

import cn.enaium.noexpensive.Config
import net.minecraft.command.AbstractCommand
import net.minecraft.command.CommandException
import net.minecraft.command.CommandSource
import net.minecraft.server.MinecraftServer
import net.minecraft.text.LiteralText
import net.minecraft.text.TranslatableText

/**
 * @author Enaium
 */
class NoExpensiveCommand : AbstractCommand() {
    override fun getCommandName(): String {
        return "noexpensive"
    }

    override fun getUsageTranslationKey(source: CommandSource): String? {
        return null
    }

    override fun method_3279(minecraftServer: MinecraftServer, commandSource: CommandSource, args: Array<String>) {
        if (args.isEmpty()) {
            commandSource.sendMessage(LiteralText("§cNoExpensive §7by §bEnaium"))
            return
        }
        if (args[0].equals("reload", ignoreCase = true)) {
            Config.load()
            commandSource.sendMessage(TranslatableText("command.reload.success"))
        }
    }
}
